services/keycloak/javascript/mappers/groups-and-roles.js

```diff
@@ -1,33 +1,50 @@
 var ArrayList = Java.type("java.util.ArrayList");
+var HashMap = Java.type("java.util.HashMap");
+var HashSet = Java.type("java.util.HashSet");
+
 var groupsAndRoles = new ArrayList();
-var forEach = Array.prototype.forEach;
+var groupProjectIds = new HashMap();
+var projectGroupProjectIds = new HashMap();
 
+var forEach = Array.prototype.forEach;
 // add all groups the user is part of
 forEach.call(user.getGroups().toArray(), function(group) {
-  // remove the group role suffixes
-  //lets check if the group has a parent if this is a child
-  if(group.getFirstAttribute("type") == "role-subgroup") {
-    var parent = group.getParent();
-    if(parent.getFirstAttribute("type") == "project-default-group") {
-        var projectIds = parent.getFirstAttribute("lagoon-projects");
-        if(projectIds !== null) {
-            forEach.call(projectIds.split(","), function(g) {
-              groupsAndRoles.add("p" + g);
-            });
-            return;
-        }
-    }
-  }
-
-  var groupName = group.getName().replace(/-owner|-maintainer|-developer|-reporter|-guest/gi,"");
-  groupsAndRoles.add(groupName);
-  return;
+	// remove the group role suffixes
+	// lets check if the group has a parent if this is a child
+	var groupName = group.getName().replace(/-owner|-maintainer|-developer|-reporter|-guest/gi,"");
```

Comment:
This should probably have an anchor on the end of the group name, and remove the `gi` flags. Otherwise couldn't users create a group in Lagoon like `i-am-the-Owner` which will mess up this logic?